src/auto-evo/mutation_strategy/CommonMutationFunctions.cs

```diff
@@ -0,0 +1,192 @@
+                        for (int rotation = 0; rotation <= 5; ++rotation)
+                        {
+                            result.Orientation = rotation;
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // We didn't find an open spot, this doesn't make much sense
+        throw new Exception("Mutation code could not find a good position " +
+            "for a new organelle");
+    }
+
+    public static void AttachIslandHexes(OrganelleLayout<OrganelleTemplate> organelles)
+    {
+        var islandHexes = organelles.GetIslandHexes();
+
+        // Attach islands
+        while (islandHexes.Count > 0)
+        {
+            var mainHexes = organelles.ComputeHexCache().Except(islandHexes);
+
+            // Compute shortest hex distance
+            Hex minSubHex = default;
+            int minDistance = int.MaxValue;
+            foreach (var mainHex in mainHexes)
+            {
+                foreach (var islandHex in islandHexes)
+                {
+                    var sub = islandHex - mainHex;
+                    int distance = (Math.Abs(sub.Q) + Math.Abs(sub.Q + sub.R) + Math.Abs(sub.R)) / 2;
+                    if (distance < minDistance)
+                    {
+                        minDistance = distance;
+                        minSubHex = sub;
+
+                        // early exit if minDistance == 2 (distance 1 == direct neighbour => not an island)
+                        if (minDistance == 2)
+                            break;
+                    }
+                }
+
+                // early exit if minDistance == 2 (distance 1 == direct neighbour => not an island)
+                if (minDistance == 2)
+                    break;
+            }
+
+            // Calculate the path to move island organelles.
+            // If statement is there because otherwise the path could be (0, 0).
+            if (minSubHex.Q != minSubHex.R)
+                minSubHex.Q = (int)(minSubHex.Q * (minDistance - 1.0) / minDistance);
+
+            minSubHex.R = (int)(minSubHex.R * (minDistance - 1.0) / minDistance);
+
+            // Move all island organelles by minSubHex
+            foreach (var organelle in organelles.Where(o => islandHexes.Any(h =>
+                         o.Definition.GetRotatedHexes(o.Orientation).Contains(h - o.Position))))
+            {
+                organelle.Position -= minSubHex;
+            }
+
+            islandHexes = organelles.GetIslandHexes();
+        }
+    }
+
+    private static int[] SideTraversalOrder(Hex hex, Direction direction, Random random)
+    {
+        if (hex.Q < 0)
+        {
+            if (direction == Direction.Front)
+            {
+                return [1];
```

Comment:
Sadly this syntax allocates memory each time, so all of these lists need to be cached in static const variables. Also this should return `Hex.HexSide` for type safety reasons as we are working with a type safe language and not JavaScript here.

Reply:
I can't seem to get this happy with making a Hex.HexSide[] static

Reply:
To me it seems like this should work:
```
private static readonly Hex.HexSide[] traversalOrder1 = [Hex.HexSide.Top];
```
(put in the right place in the class). And then `return traversalOrder1` here. At least my IDE doesn't immediately complain that shouldn't compile.